multiversx_sdk_cli/cli_wallet.py

```diff
@@ -89,38 +111,155 @@ def setup_parser(args: List[str], subparsers: Any) -> Any:
+        pubkey = secret_key.generate_public_key()
+        address = pubkey.to_address(address_hrp)
         pem_file = UserPEM(address.bech32(), secret_key)
-        pem_file_path = prepare_file(args.output_path, ".pem")
-        pem_file.save(pem_file_path)
-        logger.info(f"Pem wallet generated: {pem_file_path}")
-    if args.json:
-        json_file = prepare_file(args.output_path, ".json")
-        password = getpass.getpass("Enter a new password:")
+        pem_file.save(outfile)
+    else:
+        raise KnownError(f"Unknown format: {format}")
+
+    logger.info(f"Wallet ({format}) saved: {outfile}")
+
 
-        keyfile = convert_to_keyfile_object(secret_key.buffer, pubkey.buffer, password, None, "erd")
-        with open(output_path, 'w') as json_file:
-            dump(keyfile, json_file, indent=4)
+def convert_wallet(args: Any):
+    infile = Path(args.infile).expanduser().resolve() if args.infile else None
+    outfile = Path(args.outfile).expanduser().resolve() if args.outfile else None
+    in_format = args.in_format
+    out_format = args.out_format
+    address_index = args.address_index
+    address_hrp = args.address_hrp
 
-        logger.info(f"Json wallet generated: {json_file}")
+    if outfile and outfile.exists():
+        raise KnownError(f"File already exists, will not overwrite: {outfile}")
+
+    if infile:
+        input_text = infile.read_text()
+    else:
+        print(f"Press 'Ctrl-D' (Linux / MacOS) or 'Ctrl-Z' (Windows) when done.")
+        input_text = sys.stdin.read().strip()
+
+    mnemonic, secret_key = _load_wallet(input_text, in_format, address_index)
+    output_text = _create_wallet_content(out_format, mnemonic, secret_key, address_index, address_hrp)
+
+    if outfile:
+        outfile.write_text(output_text)
+    else:
+        print("Output:")
+        print()
+        print(output_text)
+
+
+def _load_wallet(input_text: str, in_format: str, address_index: int) -> Tuple[Optional[Mnemonic], Optional[UserSecretKey]]:
+    if in_format == WALLET_FORMAT_RAW_MNEMONIC:
+        input_text = " ".join(input_text.split())
+        mnemonic = Mnemonic(input_text)
+        return mnemonic, None
+
+    if in_format == WALLET_FORMAT_KEYSTORE_MNEMONIC:
+        password = getpass.getpass("Enter the password for the input keystore:")
+        keyfile = json.loads(input_text)
+        mnemonic = UserWallet.decrypt_mnemonic(keyfile, password)
+        return mnemonic, None
+
+    if in_format == WALLET_FORMAT_KEYSTORE_SECRET_KEY:
+        password = getpass.getpass("Enter the password for the input keystore:")
+        keyfile = json.loads(input_text)
+        secret_key = UserWallet.decrypt_secret_key(keyfile, password)
+        return None, secret_key
+
+    if in_format == WALLET_FORMAT_PEM:
+        secret_key = UserPEM.from_text(input_text, address_index).secret_key
+        return None, secret_key
+
+    raise KnownError(f"Cannot load wallet, unknown input format: {in_format}")
```

Comment:
might print all the available options, so the output would look like `Invalid wallet input format <{in_format}>. Make sure you use one of these: [raw-mnemonic, keystore-mnemonic, keystore-secret-key, pem]`

Reply:
Added. Normally, this is unreachable (since the CLI arguments parser applies the constraints beforehand).